### PR TITLE
Jupyter extension for easier development with Blocks

### PR DESCRIPTION
**Pull request by abidlabs**
Adds a jupyter extension `gradio` and associated magic command `%%blocks` that makes it easier to do iterative development with blocks, as shown in the GIF below:

Recording 2022-05-17 at 23 15 54 (1)

### CODE REVIEW

File: gradio/ipython_ext.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+import gradio
+
+
+def load_ipython_extension(ipython):
+    demo = gradio.Blocks()
+
+    @register_cell_magic
+    def blocks(line, cell):
+        if "imp" in line:
```
Comment (aliabid94) [inline gradio/ipython_ext.py:14]: what is this checking?

File: gradio/ipython_ext.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            from gradio.components import (
```
Comment (aliabid94) [inline gradio/ipython_ext.py:15]: why do we have to import everything like this? is there a more maintainable way?

File: gradio/ipython_ext.py (hunk 1 above)
Comment (abidlabs) [reply gradio/ipython_ext.py:14]: If you do %%blocks imp then it becomes even easier to do development as all of the top-level gradio classes become imported: 

image

File: gradio/ipython_ext.py (hunk 2 above)
Comment (abidlabs) [reply gradio/ipython_ext.py:15]: >Can we import everything from init.py?

Hmm I think that will create a circular import

But I'll use wildcards, I think that's pretty maintainable

Comment (abidlabs) [reply gradio/ipython_ext.py:15]: Actually nvm: 
image

**Comment by FarukOzderim** [comment]
@abidlabs  What does this PR mainly provide, could you elaborate? Does it not create a new server from scratch but update the blocks app?

**Comment by FarukOzderim** [comment]
Also lets create an issue to add this to docs if we have not.

**Pull request merged**